frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -267,8 +275,10 @@
                         // vm.upload(vm.input_file);
                     }
                     var parameters = {};
-                    parameters.url = 'jobs/challenge/' + vm.challengeId + '/challenge_phase/' + vm.phaseId + '/submission/';
-                    parameters.method = 'POST';
+                    var phase_id = vm.phaseId;
+                    var challenge_url = stringToTemplate.convert(challenge.CHALLENGE.MAKE_SUBMISSION.JOBS_CHALLENGE_ENDPOINT, {challenge_id: challenge_id});
+                    parameters.url = stringToTemplate(challenge_url, {phase_id: phase_id});
```

Comment:
Call the function `stringToTemplate.convert(*args)` instead of `stringToTemplate(*agrs)`